homeassistant/components/airvisual/__init__.py

```diff
@@ -311,6 +313,25 @@ async def async_migrate_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
                 )
             )
 
+    # 2 -> 3: Moving AirVisual Pro to its own domain
+    elif version == 2:
+        version = 3
+
+        if entry.data[CONF_INTEGRATION_TYPE] == INTEGRATION_TYPE_NODE_PRO:
+            data = {**entry.data}
+            data.pop(CONF_INTEGRATION_TYPE)
+
+            tasks = [
+                hass.config_entries.async_remove(entry.entry_id),
+                hass.config_entries.flow.async_init(
+                    AIRVISUAL_PRO_DOMAIN, context={"source": SOURCE_IMPORT}, data=data
```

Comment:
I was going to suggest integration discovery, but we already know its valid so import seems to fit better 👍